src/main/java/me/mattstudios/holovid/DisplayTask.java

```diff
@@ -58,32 +59,31 @@ public void setRunning(final boolean running) {
         this.running = running;
     }
 
-    private String[] getFrame(final File file) {
+    private IChatBaseComponent[] getFrame(final File file) {
 
         try {
             final BufferedImage image = ImageIO.read(file);
-            final String[] frame = new String[image.getHeight()];
-
-            for (int i = 0; i < image.getHeight(); i++) {
-
-                final StringBuilder builder = new StringBuilder("{\"text\":\"\",\"extra\":[");
-
-                for (int j = 0; j < image.getWidth(); j++) {
-                    final int color = image.getRGB(j, i);
-                    builder.append("{\"color\":\"#").append(String.format("%06x", color & 0x00FFFFFF)).append("\",\"text\":\"█\"},");
+            final IChatBaseComponent[] frame = new IChatBaseComponent[image.getHeight()];
+            final int width = image.getWidth();
+            final int height = image.getHeight();
+
+            final int[] rgbArray = image.getRGB(0, 0, width, height, null, 0, width);
+
+            for (int y = 0; y < height; y++) {
+                ChatBaseComponent component = new ChatComponentText("");
+                for (int x = 0; x < width; x++) {
+                    ChatComponentText text = new ChatComponentText("█");
+                    text.c(ChatModifier.b.setColor(ChatHexColor.a("#" + Integer.toHexString(rgbArray[y * width + x]).substring(2))));
```

Comment:
you should keep `String.format("%06x", color & 0x00FFFFFF)`, since toHexString doesn't keep trailing zeros

Reply:
nvm, I thought Mojang did the same stupid parsing as md